silkworm/sentry/session_sentry_client.hpp

```diff
@@ -0,0 +1,50 @@
+/*
+   Copyright 2023 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#pragma once
+
+#include <functional>
+#include <memory>
+
+#include <silkworm/infra/concurrency/coroutine.hpp>
+
+#include <boost/asio/awaitable.hpp>
+
+#include <silkworm/sentry/api/api_common/sentry_client.hpp>
+#include <silkworm/sentry/eth/status_data.hpp>
+
+namespace silkworm::sentry {
+
+class SessionSentryClientImpl;
+
+class SessionSentryClient : public api::api_common::SentryClient {
```

Comment:
I propose renaming this as `SentrySession`. Not blocking though, we can do it in another PR